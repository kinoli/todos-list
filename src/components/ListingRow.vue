<template lang="html">

  <section class="listing-row">
    <v-list-tile class="row-container" ripple @click="onClickRow(todo)">
      <v-list-tile-content>
        <v-list-tile-title class="row-title">{{ todo.title }}</v-list-tile-title>
        <v-list-tile-sub-title class="text--primary">{{ todo.user.company.name }}</v-list-tile-sub-title>
        <v-list-tile-sub-title class="row-name">{{ todo.user.name }}</v-list-tile-sub-title>
      </v-list-tile-content>

      <v-list-tile-action>
        <v-list-tile-action-text class="row-username">{{ todo.user.username }}</v-list-tile-action-text>
        <v-icon class="row-star" v-if="todo.completed" color="yellow darken-2">
          star
        </v-icon>

        <v-icon v-else class="row-star" color="grey lighten-1">
          star_border
        </v-icon>
      </v-list-tile-action>
    </v-list-tile>
  </section>

</template>

<script lang="js">
export default {
  name: 'listing-row',
  props: ['todo'],
  methods: {
    onClickRow (todo) {
      this.$router.push({ path: `/details/${todo.id}` })
    }
  }
}
</script>
