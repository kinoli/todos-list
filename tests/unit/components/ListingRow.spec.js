import Vue from 'vue'
import { shallowMount } from '@vue/test-utils'
import ListingRow from '@/components/ListingRow.vue'
import Vuetify from 'vuetify'
Vue.use(Vuetify)

const mockProps = {
  todo: {
    id: '111',
    title: 'My Title',
    completed: false,
    user: {
      name: 'Jane Doe',
      username: 'jdoe',
      company: {
        name: 'Acme Inc.'
      }
    }
  }
}

const $router = {
  push: jest.fn()
}

let wrapper
beforeEach(() => {
  wrapper = shallowMount(ListingRow, {
    propsData: mockProps
  })
})

describe('ListingRow.vue', () => {
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.find('v-list-tile-title-stub').text()).toBe(mockProps.todo.title)
    expect(wrapper.find('v-list-tile-sub-title-stub.text--primary').text()).toBe(mockProps.todo.user.company.name)
    expect(wrapper.find('.row-name').text()).toBe(mockProps.todo.user.name)
    expect(wrapper.find('.row-username').text()).toBe(mockProps.todo.user.username)
  })

  it('should have filled in star if todo IS NOT completed', () => {
    expect(wrapper.find('.row-star').text()).toBe('star_border')
    expect(wrapper.contains('.row-star[color*=grey]')).toBe(true)
  })

  it('should have an outlined star if todo IS completed', () => {
    const uncompletedProps = JSON.parse(JSON.stringify(mockProps)) // deep clone
    uncompletedProps.todo.completed = true
    wrapper.setProps(uncompletedProps)
    expect(wrapper.find('.row-star').text()).toBe('star')
    expect(wrapper.contains('.row-star[color*=yellow]')).toBe(true)
  })

  it('should push to detail page when clicked', () => {
    wrapper.vm.$router = $router
    wrapper.find('.row-container').vm.$emit('click')
    expect($router.push).toHaveBeenCalledWith({
      path: `/details/${mockProps.todo.id}`
    })
  })
})
