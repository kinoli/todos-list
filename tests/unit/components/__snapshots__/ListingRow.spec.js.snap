// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingRow.vue renders correctly 1`] = `
<section class="listing-row">
  <v-list-tile-stub activeclass="primary--text" ripple="" class="row-container">
    <v-list-tile-content-stub>
      <v-list-tile-title-stub class="row-title">My Title</v-list-tile-title-stub>
      <v-list-tile-sub-title-stub class="text--primary">Acme Inc.</v-list-tile-sub-title-stub>
      <v-list-tile-sub-title-stub class="row-name">Jane Doe</v-list-tile-sub-title-stub>
    </v-list-tile-content-stub>
    <v-list-tile-action-stub>
      <v-list-tile-action-text-stub class="row-username">jdoe</v-list-tile-action-text-stub>
      <v-icon-stub color="grey lighten-1" class="row-star">
        star_border
      </v-icon-stub>
    </v-list-tile-action-stub>
  </v-list-tile-stub>
</section>
`;
